src/dnfile/base.py

```diff
@@ -234,22 +234,63 @@ def parse(self, tables: List["ClrMetaDataTable"]):
         # if indexes
         if self._struct_indexes and tables:
             for struct_name, (attr_name, table_name) in self._struct_indexes.items():
-                table_object = None
+                table = None
                 for t in tables:
                     if t.name == table_name:
-                        table_object = t
-                if table_object:
+                        table = t
+                if table:
                     i = getattr(self.struct, struct_name, None)
-                    if i is not None and i < t.num_rows:
-                        row = t.rows[i]
-                        setattr(self, attr_name, row)
+                    if i is not None and i <= table.num_rows:
+                        setattr(self, attr_name, MDTableIndexRef(table, row_index))
```

Comment:
note: I've changed the behavior here to place a reference to the row (table, row_index) rather than the row itself. this is more consistent with the other types above. let me me know if you disagree and want this changed back.

Reply:
looks good to me